Makefile

```diff
@@ -72,6 +75,23 @@ OBJS += $(ASM_SOURCES:%.S=%.o)
 
 TARGET := kernel64.bin
 
+# Ideally we would've liked to have rule based compilation of libpfm and integration into KTF
+# However my efforts of shoving static archive into KTF binary were futile. 
+# I attempted at conditional rule based compile of project to collect object files later but
+# make doesn't allow appending or creating env vars in recipie. So we're doing conditional compile via 
+# a bash script present in tools/libpfm folder
+#
+# This is conditional env vars and conditional compiler of libpfm
+# Clean up however is unconditional and not dependent on config
+ifeq ($(CONFIG_LIBPFM), y)
+PFMLIB_NAME := libpfm
+PFMLIB_TOOLS := $(ROOT)/$(TOOLS_DIR)/$(PFMLIB_NAME)
+PFMLIB_BUILDSCRIPT := ./build_libpfm.sh
+DUMMY_RETCODE := $(shell cd $(PFMLIB_TOOLS) && $(PFMLIB_BUILDSCRIPT))
```

Comment:
Not a huge fan of this very much hidden way of calling a shell script to generate more objects.

* Can we turn this into a make rule, similar to
`libfpm.a :
    CWD=$(PFMLIB_TOOLS) $(PFMLIB_BUILDSCRIPT)
`
?

Can you also share where you failed to build the static library and link it into KTF?

Reply:
@bjoernd 
Yes that was my original plan. I created a rule and recipe
```
$(TARGET): $(PFMLIB_ARCHIVE) $(OBJS) $(PREP_LINK_SCRIPT)
	@echo "LD " $@
	$(VERBOSE) $(LD) -T $(PREP_LINK_SCRIPT) -o $@ $^
	@echo "GEN " $(SYMBOLS_NAME).S
	$(VERBOSE) $(NM) -p --format=posix $(TARGET) | $(PYTHON) $(SYMBOLS_DIR)/$(SYMBOLS_TOOL)
	@echo "CC " $(SYMBOLS_NAME).S
	$(VERBOSE) $(CC) -c -o $(SYMBOLS_NAME).o $(AFLAGS) $(SYMBOLS_NAME).S
	$(VERBOSE) rm -rf $(SYMBOLS_NAME).S
	@echo "LD " $(TARGET) $(SYMBOLS_NAME).o
	$(VERBOSE) $(LD) -T $(PREP_LINK_SCRIPT) -o $@ $(OBJS) -l$(PFMLIB_ARCHIVE).a $(SYMBOLS_NAME).o
```
But I've not been able to convince `ld` to take my archive and shove it in.

See errors below
```
ar cq libpfm.a pfmlib_common.o pfmlib_amd64.o pfmlib_intel_core.o pfmlib_intel_x86.o pfmlib_intel_x86_arch.o pfmlib_intel_atom.o pfmlib_intel_nhm_unc.o pfmlib_intel_nhm.o pfmlib_intel_wsm.o pfmlib_intel_snb.o pfmlib_intel_snb_unc.o pfmlib_intel_ivb.o pfmlib_intel_ivb_unc.o pfmlib_intel_hsw.o pfmlib_intel_bdw.o pfmlib_intel_skl.o pfmlib_intel_rapl.o pfmlib_intel_snbep_unc.o pfmlib_intel_snbep_unc_cbo.o pfmlib_intel_snbep_unc_ha.o pfmlib_intel_snbep_unc_imc.o pfmlib_intel_snbep_unc_pcu.o pfmlib_intel_snbep_unc_qpi.o pfmlib_intel_snbep_unc_ubo.o pfmlib_intel_snbep_unc_r2pcie.o pfmlib_intel_snbep_unc_r3qpi.o pfmlib_intel_ivbep_unc_cbo.o pfmlib_intel_ivbep_unc_ha.o pfmlib_intel_ivbep_unc_imc.o pfmlib_intel_ivbep_unc_pcu.o pfmlib_intel_ivbep_unc_qpi.o pfmlib_intel_ivbep_unc_ubo.o pfmlib_intel_ivbep_unc_r2pcie.o pfmlib_intel_ivbep_unc_r3qpi.o pfmlib_intel_ivbep_unc_irp.o pfmlib_intel_hswep_unc_cbo.o pfmlib_intel_hswep_unc_ha.o pfmlib_intel_hswep_unc_imc.o pfmlib_intel_hswep_unc_pcu.o pfmlib_intel_hswep_unc_qpi.o pfmlib_intel_hswep_unc_ubo.o pfmlib_intel_hswep_unc_r2pcie.o pfmlib_intel_hswep_unc_r3qpi.o pfmlib_intel_hswep_unc_irp.o pfmlib_intel_hswep_unc_sbo.o pfmlib_intel_bdx_unc_cbo.o pfmlib_intel_bdx_unc_ubo.o pfmlib_intel_bdx_unc_sbo.o pfmlib_intel_bdx_unc_ha.o pfmlib_intel_bdx_unc_imc.o pfmlib_intel_bdx_unc_irp.o pfmlib_intel_bdx_unc_pcu.o pfmlib_intel_bdx_unc_qpi.o pfmlib_intel_bdx_unc_r2pcie.o pfmlib_intel_bdx_unc_r3qpi.o pfmlib_intel_skx_unc_cha.o pfmlib_intel_skx_unc_iio.o pfmlib_intel_skx_unc_imc.o pfmlib_intel_skx_unc_irp.o pfmlib_intel_skx_unc_m2m.o pfmlib_intel_skx_unc_m3upi.o pfmlib_intel_skx_unc_pcu.o pfmlib_intel_skx_unc_ubo.o pfmlib_intel_skx_unc_upi.o pfmlib_intel_knc.o pfmlib_intel_slm.o pfmlib_intel_knl.o pfmlib_intel_knl_unc_imc.o pfmlib_intel_knl_unc_edc.o pfmlib_intel_knl_unc_cha.o pfmlib_intel_knl_unc_m2pcie.o pfmlib_intel_glm.o pfmlib_intel_netburst.o pfmlib_amd64_k7.o pfmlib_amd64_k8.o pfmlib_amd64_fam10h.o pfmlib_amd64_fam11h.o pfmlib_amd64_fam12h.o pfmlib_amd64_fam14h.o pfmlib_amd64_fam15h.o pfmlib_amd64_fam17h.o pfmlib_amd64_fam16h.o
make[2]: Leaving directory '/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/github/KTF/ktf/third-party/libpfm/lib'
make[1]: Leaving directory '/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/github/KTF/ktf/third-party/libpfm'
find /home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/github/KTF/ktf/third-party/libpfm -name \*.c -delete
find /home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/github/KTF/ktf/third-party/libpfm -name \*.c -delete
LD  kernel64.bin
GEN  symbols.S
CC  symbols.S
LD  kernel64.bin symbols.o
ld: cannot find -l/home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/github/KTF/ktf/third-party/libpfm/libpfm.a
Makefile:113: recipe for target 'kernel64.bin' failed
make: *** [kernel64.bin] Error 1

dkgupta@uaf44af6b6cb95f:~/data_drive1/sources/github/KTF/ktf$ ls /home/ANT.AMAZON.COM/dkgupta/data_drive1/sources/github/KTF/ktf/third-party/libpfm
config.mk  COPYING  include  lib  libpfm-4.10.1.tar.gz  _**libpfm.a**_  Makefile  README  rules.mk
```
I've almost tried every option of using `-L` to specify directory and then give only library name using `-l`.
So far I've not been successful. Any help here would be appreciated and I would love to use rule based recipe for build integration.